### PR TITLE
feat(clean): add clean_address function

### PR DESCRIPTION
**Pull request by ryanwdale**
# Description

Implements the `clean_address()` and `validate_address()` functions for cleaning messy addresses.

# How Has This Been Tested?

Unit tests are provided and snapshots are below.

# Snapshots:

### Default clean_address

Screen Shot 2021-02-03 at 2 53 53 PM

### `split` parameter

Screen Shot 2021-02-03 at 2 54 06 PM

### `output_format` parameter 

Screen Shot 2021-02-03 at 2 54 15 PM

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have already squashed the commits and make the commit message conform to the project standard.
- [ ] I have already marked the commit with "BREAKING CHANGE" or "Fixes #" if needed.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: dataprep/clean/clean_address.py [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+        A tuple containing parts of the address that must be included for the
+         address to be successfully cleaned
+    clean
+        If True, a tuple (index, status) is returned.
+        If False, the function returns True/False to be used by the validate address function.
+    """
+    if address_str in NULL_VALUES:
+        return (None, "null") if clean else False
+
+    address_str = address_str.replace("(", "")
```
Comment (brandonlockhart) [inline dataprep/clean/clean_address.py:275]: What do you think about `address_str = re.sub(r"[().]", "", address_str)` here? Maybe it's a bit more concise.

File: dataprep/clean/clean_address.py [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+
+def _get_column_names(output_format: str, split: bool) -> List[str]:
+    """
+    returns the column names that will be present in the final dataframe,
+    based on the output_format
+    """
+    if not split:
+        return [name.strip() for name in output_format.split("\t")]
+
+    keywords = [
```
Comment (brandonlockhart) [inline dataprep/clean/clean_address.py:374]: Can this be made a global variable?

File: dataprep/clean/clean_address.py [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+
+    address_str = address_str.replace("(", "")
+    address_str = address_str.replace(")", "")
+    address_str = address_str.replace(".", "")
+
+    try:
+        address, _ = usaddress.tag(address_str, TAG_MAPPING)
+
+    except usaddress.RepeatedLabelError:
+        return None, "unknown"
```
Comment (brandonlockhart) [inline dataprep/clean/clean_address.py:283]: Should this be `return (None, "unknown") if clean else False`?

**Review by brandonlockhart — CHANGES_REQUESTED** [review]
Excellent PR Ryan! I just left a couple of minor comments.

Can you also please follow the instructions [here](https://github.com/sfu-db/dataprep/wiki/Docstring-Best-Practice#auto-generating-the-api-reference) and add the modified file to this PR?

After these changes, I think it can be merged.

**Comment by ryanwdale** [comment]
Thanks Brandon, I agree with all your comments and I've made the requested changes.

**Review by brandonlockhart — APPROVED** [review]
LGTM

**Pull request merged**